### PR TITLE
fix(authenticator): retry token refresh on error

### PR DESCRIPTION
**Pull request by velrest**
We retry the refresh if:
- The server returns any 500 status code
- The request does not resolve and an error is thrown because the network connection is broken

### CODE REVIEW

**Comment by anehx** [comment]
Please document those config options

File: addon/authenticators/oidc.js [hunk 1]
```diff
@@ -102,19 +105,35 @@ export default BaseAuthenticator.extend({
...
+          redirect_uri: this.redirectUri
+        }
+      });
+      return this._handleAuthResponse(data);
+    } catch (e) {
+      if (
+        (isServerError(e) || isAbortError(e) || isTimeoutError(e)) &&
+        retryCount < amountOfRetries - 1
+      ) {
+        run.later(
```
Comment (anehx) [inline addon/authenticators/oidc.js]: Import `later` directly: `import { later } from "@ember/runloop"`

File: addon/config.js [hunk 2]
```diff
@@ -15,7 +15,11 @@ export default Object.assign(
     refreshLeeway: 1000 * 30,
     tokenPropertyName: "access_token",
     authHeaderName: "Authorization",
-    authPrefix: "Bearer"
+    authPrefix: "Bearer",
+    // How many times should a token refresh be retried if a request fails
+    amountOfRetries: 3,
+    // How long should be waited between the retries
```
Comment (anehx) [inline addon/config.js]: The comments are unnecessary since you documented it properly now

File: addon/authenticators/oidc.js (hunk 1 above)
Comment (velrest) [reply addon/authenticators/oidc.js]: done

File: addon/config.js (hunk 2 above)
Comment (velrest) [reply addon/config.js]: done

**Comment by czosel** [comment]
Just a thought: Wouldn't this be the perfect use-case for exponential backoff? Then you could remove at least one, probably both config options.

**Comment by velrest** [comment]
@czosel im not sure about the exponential part but a constant backoff would make sense in this case.
Since the token refresh request ist not really resource heavy, i dont think a exponential backoff is needed.

**Comment by czosel** [comment]
Sounds fair! Either way, it would be nice to reduce the amount of config options.

**Pull request merged**